frameworks/elastic/src/main/dist/svc.yml

```diff
@@ -386,6 +398,99 @@ pods:
+    {{#ENABLE_VIRTUAL_NETWORK}}
+    networks:
+      {{VIRTUAL_NETWORK_NAME}}:
+        labels: {{VIRTUAL_NETWORK_PLUGIN_LABELS}}
+    {{/ENABLE_VIRTUAL_NETWORK}}
+    secrets:
+      {{#TASKCFG_ALL_SG_INTERNAL_USERS_SECRET_PATH}}
+      sg_internal_users:
+        secret: {{TASKCFG_ALL_SG_INTERNAL_USERS_SECRET_PATH}}
+        file: sg_internal_users.yml
+      {{/TASKCFG_ALL_SG_INTERNAL_USERS_SECRET_PATH}}
+      {{#TASKCFG_ALL_SG_CONFIG_SECRET_PATH}}
+      sg_config:
+        secret: {{TASKCFG_ALL_SG_CONFIG_SECRET_PATH}}
+        file: sg_config.yml
+      {{/TASKCFG_ALL_SG_CONFIG_SECRET_PATH}}
+      {{#TASKCFG_ALL_SG_ACTION_GROUPS_SECRET_PATH}}
+      sg_action_groups:
+        secret: {{TASKCFG_ALL_SG_ACTION_GROUPS_SECRET_PATH}}
+        file: sg_action_groups.yml
+      {{/TASKCFG_ALL_SG_ACTION_GROUPS_SECRET_PATH}}
+      {{#TASKCFG_ALL_SG_ROLES_SECRET_PATH}}
+      sg_roles:
+        secret: {{TASKCFG_ALL_SG_ROLES_SECRET_PATH}}
+        file: sg_roles.yml
+      {{/TASKCFG_ALL_SG_ROLES_SECRET_PATH}}
+      {{#TASKCFG_ALL_SG_ROLES_MAPPING_SECRET_PATH}}
+      sg_roles_mapping:
+        secret: {{TASKCFG_ALL_SG_ROLES_MAPPING_SECRET_PATH}}
+        file: sg_roles_mapping.yml
+      {{/TASKCFG_ALL_SG_ROLES_MAPPING_SECRET_PATH}}
+      {{#TASKCFG_ALL_SG_TENANTS_SECRET_PATH}}
+      sg_tenants:
+        secret: {{TASKCFG_ALL_SG_TENANTS_SECRET_PATH}}
+        file: sg_tenants.yml
+      {{/TASKCFG_ALL_SG_TENANTS_SECRET_PATH}}
+    uris:
+      - {{ELASTICSEARCH_JAVA_URI}}
+      - {{ELASTICSEARCH_URI}}
+      - {{SEARCHGUARD_SGADMIN_URI}}
+    placement:
+    ports:
+      sgadmin-http:
+        port: 0
+        env-key: PORT_HTTP
+        advertise: true
+        vip:
+          prefix: sgadmin
+          port: 9200
+      sgadmin-transport:
+        port: 0
+        env-key: PORT_TRANSPORT
+        advertise: true
+        vip:
+          prefix: sgadmin
+          port: 9300
+    tasks:
+      admin:
+        goal: RUNNING
+        cpus: 0.25
+        memory: 512
+        cmd: |
+         set -e
+         export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/)
+         ./elasticsearch-{{ELASTIC_VERSION}}/bin/elasticsearch-plugin install --batch file://$(ls $MESOS_SANDBOX/search-guard-6-*.zip)
+         for sg_config_file in sg_internal_users.yml sg_action_groups.yml sg_roles.yml sg_roles_mapping.yml sg_config.yml sg_tenants.yml; do
+           if [ -f $MESOS_SANDBOX/$sg_config_file ]; then
+             ln -sf $MESOS_SANDBOX/$sg_config_file sgconfig/$sg_config_file
+           fi
+         done
+         chmod +x elasticsearch-{{ELASTIC_VERSION}}/plugins/search-guard-6/tools/sgadmin.sh
+
+         cat > sgadmin.sh << 'EOF'
+         #!/bin/bash
+         openssl pkcs8 -in node.key -topk8 -nocrypt -out node.key.pkcs8
+         elasticsearch-{{ELASTIC_VERSION}}/plugins/search-guard-6/tools/sgadmin.sh -key node.key.pkcs8 -cert node.crt -cacert node.ca -cd sgconfig -cn elastic -h master.${FRAMEWORK_NAME//\//}.l4lb.thisdcos.directory
+         EOF
```

Comment:
Hm, I'm a little on the fence about this. What I'd like to suggest is rather than creating a script every time the task launches, can we save this to a file in `downloads.mesosphere`? Since `FRAMEWORK_NAME` is dynamic, we could add a placeholder to `sed` the name instead during task launch. 

But the code is so concise, maybe it doesn't make sense to. I'll defer to @mpereira on this one.

Reply:
Thanks for the idea!
But I am missing something, can you please clarify this more? 
Isn't creating the script costs almost nothing and we will just add another level of indirection?

Reply:
Yeah that's why I'm on the fence. Doesn't seem worth the additional level.